connectors/bigquery/flink-connector-gcp-bigquery/src/main/java/com/google/flink/connector/gcp/bigquery/BigQueryCatalog.java

```diff
@@ -0,0 +1,440 @@
+import com.google.api.services.bigquery.BigqueryScopes;
+import com.google.api.services.bigquery.model.Dataset;
+import com.google.api.services.bigquery.model.DatasetList;
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableList;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+
+/**
+ * Flink Catalog Support for BigQuery. This class provides the implementation
+ * for Catalog operations for BigQuery.
+ */
+public class BigQueryCatalog extends AbstractCatalog {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BigQueryCatalog.class);
+
+    private Bigquery client;
+
+    public static final String DEFAULT_DATASET = "default";
+
+    private final String projectId;
+
+    public BigQueryCatalog(String catalogName, String defaultDataset, String project, String credentialFile) {
+        super(catalogName, defaultDataset);
+        this.projectId = project;
+        LOG.info("Created BigQueryCatalog: {}", catalogName);
+    }
+
+    @Override
+    public void open() throws CatalogException {
+        LOG.info("Connected to BigQuery with project ID: {}", this.projectId);
+
+        try {
+            HttpCredentialsAdapter httpCredentialsAdapter
+                    = new HttpCredentialsAdapter(
+                            GoogleCredentials.getApplicationDefault().createScoped(BigqueryScopes.all()));
+
+            this.client = new Bigquery.Builder(
+                    GoogleNetHttpTransport.newTrustedTransport(),
+                    GsonFactory.getDefaultInstance(),
+                    httpRequest -> {
+                        httpCredentialsAdapter.initialize(httpRequest);
+                        httpRequest.setThrowExceptionOnExecuteError(false);})
+                    .setApplicationName("BigQuery Flink Catalog Plugin")
+                    .build();
+        } catch (GeneralSecurityException | IOException e) {
+            throw new CatalogException("Failed to create BigQuery client", e);
+        }
+    }
+
+    @Override
+    public void close() throws CatalogException {
+        if (client != null && client.getRequestFactory().getTransport() instanceof com.google.api.client.http.javanet.NetHttpTransport) {
+            try {
+                this.client.getRequestFactory().getTransport().shutdown();
+                LOG.info("Shutdown the underlying HTTP transport for BigQuery client.");
+            } catch (IOException e) {
+                throw new CatalogException("Failed to shutdown the HTTP transport", e);
+            } finally {
+                client = null;
+            }
+        } else if (client != null) {
+            client = null;
+            LOG.info("BigQuery client was open, but using a transport without explicit shutdown. Nullifying client.");
+        } else {
+            LOG.info("BigQuery client was not open, nothing to close.");
+        }
```

Comment:
this can just be one final client = null. we would lose the log message but probably worth it for code simplicity